src/main/java/org/kuali/kra/budget/document/BudgetDocumentRule.java

```diff
@@ -154,7 +154,86 @@ protected boolean processCustomSaveDocumentBusinessRules(Document document) {
     * @param budgetDocument
     * @return
     */
-    @SuppressWarnings("rawtypes")
+    @SuppressWarnings("rawtypes")  
+	protected boolean processBudgetProjectIncomeBusinessRule(BudgetDocument budgetDocument) {
+        boolean valid = true;
+        MessageMap errorMap = GlobalVariables.getMessageMap();
+        int i = 0;
+        
+        // Get each budget cost share.
+        for (BudgetCostShare budgetCostShare : budgetDocument.getBudget().getBudgetCostShares()) {
+            String errorPath = "budgetCostShare[" + i + "]";
+            errorMap.addToErrorPath(errorPath);
+            
+            // Validate share percentage.
+            if(budgetCostShare.getSharePercentage() != null && (budgetCostShare.getSharePercentage().isLessThan(new BudgetDecimal(0)) || 
+               budgetCostShare.getSharePercentage().isGreaterThan(new BudgetDecimal(100)))) {
+                errorMap.putError("sharePercentage", KeyConstants.ERROR_COST_SHARE_PERCENTAGE);
+                valid = false;
+            }
+            
+            // Check for duplicate fiscal year, source account and unit number on all unchecked cost shares.
+            if (i < budgetDocument.getBudget().getBudgetCostShareCount()) {
+                for (int j = i + 1; j < budgetDocument.getBudget().getBudgetCostShareCount(); j++) {
+                    BudgetCostShare tmpCostShare = budgetDocument.getBudget().getBudgetCostShare(j);
+                    int thisFiscalYear = budgetCostShare.getProjectPeriod() == null ? Integer.MIN_VALUE : budgetCostShare.getProjectPeriod();
+                    int otherFiscalYear = tmpCostShare.getProjectPeriod() == null ? Integer.MIN_VALUE : tmpCostShare.getProjectPeriod();
```

Comment:
Why the unboxing w/ the two Integer's here? You wouldn't then have to use the undesirable (thisFilscalYear + "") trick later for the error map strings.